rollup-interface/src/state_machine/optimistic.rs

```diff
@@ -0,0 +1,46 @@
+//! Utilities for building an optimistic state machine
+use borsh::{BorshDeserialize, BorshSerialize};
+use serde::{Deserialize, Serialize};
+
+use crate::zk::StateTransition;
+
+/// A proof that the attester was bonded at the transition num `transition_num`.
+/// For rollups using the `jmt`, this will be a `jmt::SparseMerkleProof`
+#[derive(
+    Debug, Clone, PartialEq, Eq, BorshSerialize, BorshDeserialize, Serialize, Deserialize, Default,
+)]
+pub struct ProofOfBond<StateProof: BorshSerialize> {
+    /// The transition number for which the proof of bond applies
+    pub transition_num: u64,
```

Comment:
Can't we recover the transition number from the `da_block_hash`?  Having the attester post it feels like an easy way to confuse trusted for untrusted data. If not, maybe we should rename it to `claimed_transition_num` so that we remember that it might be falsified.

Reply:
We need to have the transition number to recover the root hash from the chain state. We would be able to store the block hash and recover it from the chain state if needed, but I felt that the transition number is more straightforward. 
This is basically a hint that is used to verify the root hash claimed by the proof, we can definitely rename it `claimed_transition_num`